lib/algo/AdaptiveDMD.h

```diff
@@ -0,0 +1,126 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2021, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Description: Computes the AdaptiveDMD algorithm on the given snapshot matrix.
+//              The AdaptiveDMD algorithm should be used if the dt changes
+//              between the samples. This algorithm interpolates the ununiformly
+//              sampled samples such that they are uniformly distanced and the
```

Comment:
This could be written more clearly, for example: "This algorithm uniformly interpolates the samples that may have been taken with variable steps, using the constant step dt."